### DISCUSSION TITLE
Ternary void

### DISCUSSION THREAD

**Discussion by NN---** [dotnet/csharplang#6437] [Language Ideas] [answered]
C# language doesn't allow using ternary operator (?:) for void type.
This is inconvenient with expression body methods.
The proposal is to allow it making language more consistent.

`void` can return `void`
```cs
void G() {}    
void F() => G();  // OK
```

`non void` can use ternary with returning `non void`
```cs
int W() => 1;
int Q() => true ? W() : W(); // OK
```

`void` cannot use ternary to invoke `void`
```cs
void G() {}    
void F() => true ? G() : G(); // Error

void F2() { if (true) G() else G(); } // Only bodied method
```

**Comment by huoyaoyuan** [comment, marked as answer]
Duplicate of #6013 (And I remember there should be more).

**Reply by NN---** [reply]
Thanks. Couldn't find it.
